### ISSUE TITLE
rsync backup weird behaviour - sends id_rsa_mid to the remote server

### ISSUE THREAD

**Issue by Bertrand256** [mail-in-a-box/mailinabox#1621]
Hi,
I'm testing rsync backups with the latest miab version and I've faced the same issues as those described by virakocha in this report: https://discourse.mailinabox.email/t/help-restoring-a-backup/5245

The biggest problem for me is that the rsa private key (`id_rsa_miab` ) is copied to the remote side which is probably not very desired behaviour.

After creating the first backup, the backup folder on the remote side has the following structure:
```
├── [drwxrwxr-x]  duplicity-full.20190808T135731Z.manifest.gpg
│   ├── [-rw-------]  duplicity-full.20190808T135731Z.manifest.gpg
│   └── [-rw-------]  id_rsa_miab
├── [drwxrwxr-x]  duplicity-full.20190808T135731Z.vol1.difftar.gpg
│   ├── [-rw-------]  id_rsa_miab
│   └── [-rw-------]  mktemp-JfRFrV-2
├── [drwxrwxr-x]  duplicity-full-signatures.20190808T135731Z.sigtar.gpg
│   ├── [-rw-------]  duplicity-full-signatures.20190808T135731Z.sigtar.gpg
│   └── [-rw-------]  id_rsa_miab
└── [-rw-------]  id_rsa_miab
```
 
Note, that the actual data files are created inside folders, which causes `IOError: [Errno 21] Is a directory` errors when recovering data, mentioned by virakocha.

I've dug a little deeper and after enabling logging I've came to some conclusions. 
The problem is caused by the following rsync commands, being executed by duplicity:
```
Writing duplicity-full.20190808T144045Z.vol1.difftar.gpg
Reading results of 'rsync -e 'ssh  -oBatchMode=yes '-i /root/.ssh/id_rsa_miab'' "-e /usr/bin/ssh -oStrictHostKeyChecking=no -oBatchMode=yes -p 22 -i /root/.ssh/id_rsa_miab" /tmp/duplicity-L3WSTU-tempdir/mktemp-xN7nU1-2 test@hostname:/home/test/backup3/duplicity-full.20190808T144045Z.vol1.difftar.gpg'
Deleting /tmp/duplicity-L3WSTU-tempdir/mktemp-xN7nU1-2
AsyncScheduler: task completed successfully
Processed volume 1
Writing duplicity-full-signatures.20190808T144045Z.sigtar.gpg
Reading results of 'rsync -e 'ssh  -oBatchMode=yes '-i /root/.ssh/id_rsa_miab'' "-e /usr/bin/ssh -oStrictHostKeyChecking=no -oBatchMode=yes -p 22 -i /root/.ssh/id_rsa_miab" /home/user-data/backup/cache/e443d84e73f169bb77a4312f79d98114/duplicity-full-signatures.20190808T144045Z.sigtar.gpg test@hostname:/home/test/backup3/duplicity-full-signatures.20190808T144045Z.sigtar.gpg'
Deleting /home/user-data/backup/cache/e443d84e73f169bb77a4312f79d98114/duplicity-full-signatures.20190808T144045Z.sigtar.gpg
Writing duplicity-full.20190808T144045Z.manifest.gpg
Reading results of 'rsync -e 'ssh  -oBatchMode=yes '-i /root/.ssh/id_rsa_miab'' "-e /usr/bin/ssh -oStrictHostKeyChecking=no -oBatchMode=yes -p 22 -i /root/.ssh/id_rsa_miab" /home/user-data/backup/cache/e443d84e73f169bb77a4312f79d98114/duplicity-full.20190808T144045Z.manifest.gpg test@hostname:/home/test/backup3/duplicity-full.20190808T144045Z.manifest.gpg'
Deleting /home/user-data/backup/cache/e443d84e73f169bb77a4312f79d98114/duplicity-full.20190808T144045Z.manifest.gpg
Reading results of 'rsync -e 'ssh  -oBatchMode=yes '-i /root/.ssh/id_rsa_miab'' "-e /usr/bin/ssh -oStrictHostKeyChecking=no -oBatchMode=yes -p 22 -i /root/.ssh/id_rsa_miab" test@hostname:/home/test/backup3/'
```
Note that the identity_file string (`-i /root/.ssh/id_rsa_miab`) is passed twice in two instances of the `-e ssh` parameter and apparently this is the source of the problem. The first occurrence is probably converted by duplicity from this part of miab sources: https://github.com/mail-in-a-box/mailinabox/blob/e37768ca8680106b037822975445de88e3e0da14/management/backup.py#L18

After removing that line from the sources, the backup procedure behaves as expected and rsync calls look like this:
```
Writing duplicity-full-signatures.20190808T144208Z.sigtar.gpg
Reading results of 'rsync -e 'ssh  -oBatchMode=yes ' "-e /usr/bin/ssh -oStrictHostKeyChecking=no -oBatchMode=yes -p 22 -i /root/.ssh/id_rsa_miab" /home/user-data/backup/cache/e443d84e73f169bb77a4312f79d98114/duplicity-full-signatures.20190808T144208Z.sigtar.gpg test@hostname:/home/test/backup3/duplicity-full-signatures.20190808T144208Z.sigtar.gpg'
Deleting /home/user-data/backup/cache/e443d84e73f169bb77a4312f79d98114/duplicity-full-signatures.20190808T144208Z.sigtar.gpg
```
The second line:
https://github.com/mail-in-a-box/mailinabox/blob/e37768ca8680106b037822975445de88e3e0da14/management/backup.py#L19
seems to be enough to authenticate on the remote side. 

Versions:
- duplicity: 0.7.17
- rsync: 3.1.2

**Comment by jvolkenant** [comment]
I believe this was fixed in c4cb828f6562ad03497e64acd2ebb0af7294d36c?

**Comment by Bertrand256** [comment]
Closing the issue.
